pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/kube_ingress_reconciler.go

```diff
@@ -271,7 +271,7 @@ func (r *RawIngressReconciler) Reconcile(isvc *v1beta1.InferenceService) error {
 	if r.ingressConfig.IngressDomain == constants.ClusterLocalDomain {
 		isInternal = true
 	}
-	if !isInternal {
+	if !isInternal && !r.ingressConfig.DisableIngressCreation {
```

Comment:
This works nicely on clean clusters. However, I wonder how much it should be supported flipping configurations.

So, if you start with `disableIngressCreation=false` the controller will create ingresses for newly created ISVCs.

If you later flip it to `disableIngressCreation=true`, the controller will no longer create ingresses for new ISVCs. However, the ingresses for older ISVC will persist and they will no longer be updated. My rationale is that the controller should remove all managed ingresses if the setting is flipped, in order to reach the desired state.

Weirdly, if you start with `disableIngressCreation=true` and later flip to `false`, the controller will create/update all missing Ingresses for older ISVCs at startup. So, behavior is not that consistent.

**This is a non-blocking comment**